activation/wire/malfeasance.go

```diff
@@ -0,0 +1,44 @@
+package wire
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/signing"
+)
+
+//go:generate scalegen
+
+// ProofType is an identifier for the type of proof that is encoded in the ATXProof.
+type ProofType byte
+
+const (
+	DoublePublish ProofType = iota + 1
+	DoubleMarry
+)
+
+type ATXProof struct {
+	// ProofType is the type of proof that is being provided.
+	ProofType ProofType
+	// Certificates is a slice of marriage certificates showing which identities belong to the same marriage set as
+	// the one proven to be malfeasant. Up to 1024 can be put into a single proof, since by repeatedly marrying other
+	// identities there can be much more than 256 in a malfeasant marriage set. Beyond that a second proof could be
+	// provided to show that additional identities are part of the same malfeasant marriage set.
+	Certificates []ProofCertificate `scale:"max=1024"`
+	// Proof is the actual proof. Its type depends on the ProofType.
+	Proof []byte `scale:"max=1048576"` // max size of proof is 1MiB
+}
+
+// ProofCertificate proofs that two identities are part of the same marriage set.
+type ProofCertificate struct {
+	// Target of the certificate, i.e. the identity that signed the ATX containing the original certificate.
+	Target types.NodeID
+	// ID is the identity that signed the certificate.
+	ID types.NodeID
+	// Signature is the signature of the certificate, i.e. ID signed with the key of SmesherID.
+	Signature types.EdSignature
+}
```

Comment:
Having `Target` in each certificate is quite redundant as in most cases, all certs will share it. How about structure like this:
```go
type MarriageCertificates {
	Target types.NodeID
	Certificates []MarriageCertificate `scale:"max=256"`
}

type ATXProof struct {
	ProofType ProofType
	Certificates []MarriageCertificates `scale:"max=2"`
	Proof []byte `scale:"max=1048576"`
}
```